### ISSUE TITLE
[ENH] ensure that all estimators have two test parameter sets

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#3429]
We should ensure that all estimators (that have parameters) possess at least two test parameter sets.

The two (or more) parameter sets should:

* be fast to run together - `fit` is the bottleneck (so we should not overdo it with too many parameter sets)
* cover substantially different settings for all the important parameters, i.e., substantially different typical cases and/or important edge cases

Recipe:

1. search for estimators which have parameters but only a single test parameter set. These are estimators with no `get_test_params` implemented, or `get_test_params` returning only a single dictionary instead of a list of two (or more) dictionaries.
2. come up with a parameter set satisfying the above constraints and add it to the return (should be list of two or more dictionaries)

An example PR that adds second parameter sets for some estimators can be found here: https://github.com/alan-turing-institute/sktime/pull/3428

Finding some estimators that have only one parameter set can also be done speedily by using this PR https://github.com/alan-turing-institute/sktime/pull/2862 which adds a test for two parameter sets - either run the test suite from the branch locally, or look into the failing CI.

**Comment by Abelarm** [comment]
Hi @fkiraly 

I’d like to to tackle this if it’s ok for you.

**Comment by fkiraly** [comment]
sure, @Abelarm - pick an estimator!

**Comment by janpipek** [comment]
I am picking SARIMAX.

**Comment by julia-kraus** [comment]
which estimators are still left?

**Comment by fkiraly** [comment]
@julia-kraus, the failures in this diagnostic PR https://github.com/sktime/sktime/pull/2862 correspond to the ones that do have only one - it might not be 100% up to date, I'll restart it so it is:

**Comment by fkiraly** [comment]
I would like to add the test params for the `HampelFilter`, that sounds interesting.

**Comment by iaryangoyal** [comment]
I have worked on Rocket regression #6134

**Comment by namita0210** [comment]
Hi , can I take up this issue for the estimator:  "TimeSeriesForestClassier"
@fkiraly

**Comment by fkiraly** [comment]
@namita0210, absolutely! All yours!

**Comment by MMTrooper** [comment]
Hi, I will try to tackle the MatrixProfileClassifier. @fkiraly

**Comment by fkiraly** [comment]
great, thanks, @MMTrooper!

**Comment by shankariraja** [comment]
Hi @fkiraly,

I'm currently working on adding new test parameter sets for the estimators identified in the issue. I'll be focusing on ``TimeSeriesKMeansTslearn``.
I'll create a pull request once I've completed the changes and tests. In the meantime, please let me know if you have any suggestions.

Thanks!

**Comment by KaustubhUp025** [comment]
Hello @fkiraly , I will try to work on the estimator:- RocketRegressor

**Comment by Z-Fran** [comment]
Hi @fkiraly , I will try to work on `KNeighborsTimeSeriesRegressor`.

**Comment by shlok191** [comment]
Hello @fkiraly,
Could you please work on the the `LSTMFCNNetwork`? 

Thank you!!

**Comment by fkiraly** [comment]
do you mean, you would like to work on `LSTMFCNNetwork`, or are you asking me to?

**Comment by shlok191** [comment]
@fkiraly, Oh I'm sorry about that! I meant if **I** could work on this! I wrote that while traveling, so sorry again!

**Comment by fkiraly** [comment]
Sure! No worries, and thanks for contributing!

For this estimator, kindly be aware of:

* failures reported in https://github.com/sktime/sktime/issues/6153 which you might encounter
* the requirement to remove the test skip after fixing, as described in #6153 (or tests will not run)

**Comment by shlok191** [comment]
@fkiraly, Thank you so much for letting me help out!
I'll keep my eye on the possible failures and I'll remove the test skips as well :)

**Comment by shlok191** [comment]
Hello @fkiraly,

I hope that you're having a good start to your week! I wanted to let you know that I added 2 test parameters for the `LSTMFCNNetwork` [here](https://github.com/shlok191/sktime/blob/lstm_fcnn_networks/sktime/networks/lstmfcn.py). I also checked `tests/_config.py` to make sure that this estimator is included in CI tests. ☺️ 

I am really excited to test out the test parameters and getting your feedback! I can try to test out the changes locally first if that is the preferred protocol. I learned a lot about LSTMs in the context of time series from this. I would really love to possibly contribute more after this estimator's parameters are completed if that is okay.

Thank you so much again!

**Comment by fkiraly** [comment]
Great, @shlok191!

I recommend you open a pull request, where core developers can discuss your contribution further and possiby merge it!

**Comment by shlok191** [comment]
@fkiraly, I just added a PR. Thanks a lot again for letting me contribute! 😃

**Comment by fkiraly** [comment]
`sktime` is an open project, so everyone can contribute!

Thanks for your contribution!

**Comment by sanskarmodi8** [comment]
Hey @fkiraly, can I work on BaggingForecaster for this issue?

**Comment by sanskarmodi8** [comment]
Hey @fkiraly, can I work on `ColumnTransformer` for this issue?

**Comment by tajir0** [comment]
Hi @fkiraly, I have worked on `STRAY` and added a PR #7420  for it. I am really looking forward to getting your feedback and hopefully having this PR merged!

**Comment by TanviPooranmal** [comment]
Hola @fkiraly! I'd like to take up `HOG1DTransformer`.

**Comment by Adarsh2345** [comment]
Hey @fkiraly,I am working on `BaggingForecaster` for this issue

**Comment by satvshr** [comment]
> Finding some estimators that have only one parameter set can be done quickly by checking the variable EXCLUDED_TESTS_BY_TEST in sktime.tests._config, estimators which do not comply with the "number of test parameter sets" requirements are skipping "test_get_test_params_coverage"
@fkiraly the estimators in the above list and the ones generated by running the code mentioned in the issue do not match, is it because the list has not been updated yet or is there something I am missing? Thanks!

**Comment by fkiraly** [comment]
Copying my answer from discord:

We have probably forgotten in some instances to update the `EXCLUDED_TESTS_BY_TEST` variable! It might be a good idea to add a test that would ensure they are up to date.

However, there is a caveat, as there is another edge case to take care of: if the estimator has no parameters except those in the tag `reserved_params`, then it is fine to have an empty `get_test_params`, this is not covered by the code.

The actual condition should have `len(x[1].get_param_names()) - len(x[1].get_tag("reserved_params") ) > 0]` (and coerce `None` to an empty list)

**Comment by Vedant-Kaushik** [comment]
Hey @fkiraly,I am working on DummyRegressor for this issue

**Comment by Priyanshu1303d** [comment]
Hey @fkiraly , I have created a PR on `TimeBinner` related to this issue

**Comment by anoodsha** [comment]
Hi @fkiraly, I would like to contibute to this issue as a new contributor to open source

**Comment by fkiraly** [comment]
@anoodsha, welcome to sktime!

Pick an estimator to work on (see list above)

**Comment by anoodsha** [comment]
@fkiraly Hi, thank you , i would like to work on the following estimators from the list above (A-C): 
Aggregator

BaggingForecaster

CNNNetwork

ClaSPTransformer

ColumnEnsembleClassifier

ColumnTransformer

ColumnwiseTransformer

ComposableTimeSeriesForestRegressor

ConstraintViolation

CutoffSplitter

However, I noticed the list of estimators when running the same code provided above from sktime.registry looks different from the estimators provided above. thank you

**Comment by fkiraly** [comment]
> However, I noticed the list of estimators when running the same code provided above from sktime.registry looks different from the estimators provided above. thank you

Thanks - we should update the list, maybe it is outdated.

**Comment by thearshkumar** [comment]
Hello @fkiraly 

I want to take:-

LTSFDLinearForecaster
LTSFLinearForecaster
LTSFNLinearForecaster

While these are not present in the list you posted with this PR, they are present in the list generated by the code you posted above.

**Comment by fkiraly** [comment]
great, thanks!

**Comment by IsaiasRT** [comment]
Hello @fkiraly 

As soon as I fix my PR, can I contribute with this one too?

I want to take these:

- MiniRocketMultivariate
- MiniRocketMultivariateVariable
- MultiRocket

**Comment by Udayan853** [comment]
Hi! 
I noticed `ClaSPTransformer` was picked, but the last activity from the original poster was a few months ago.
I’d like to work on it if that’s okay.

**Comment by leonicus** [comment]
Hi, I'd like to work on MeanAbsoluteError and MeanAbsoluteError

**Comment by fkiraly** [comment]
> Hi! I noticed `ClaSPTransformer` was picked, but the last activity from the original poster was a few months ago. I’d like to work on it if that’s okay.

yes, that's ok, thanks!

**Comment by Priyanshu1303d** [comment]
Hey @fkiraly,
can i work on more than one estimators for this issue ? 
If yes, then should i make a separate pr for each one ??

**Comment by fkiraly** [comment]
@Priyanshu1303d, yes; for your first PR, it would be great if you deal only with a single estimator to make it easier to pick up any issues with code formatting, testing, etc.

**Comment by Abelarm** [comment]
Hi @fkiraly,

3 years after back to this issue (thanks to hacktoberfests).
Do you mind if I take one? 
Maybe I'll be more lucky?

**Comment by Priyanshu1303d** [comment]
Hi @fkiraly, I would like to work on ReducerTransform

**Comment by fkiraly** [comment]
> Hi [@fkiraly](https://github.com/fkiraly),
> 
> 3 years after back to this issue (thanks to hacktoberfests). Do you mind if I take one? Maybe I'll be more lucky?

absolutely! Let us know on discord or dev-chat if you need help getting set up. Looks good though!

**Comment by Abelarm** [comment]
How do I access dev-chat? I'll join discord

Anyway I already opened 2 PRs related to this issue.
I need a bit of help with #8916

**Comment by leonicus** [comment]
@fkiraly https://github.com/sktime/sktime/pull/8924 is ready, please lmk if something is missing :)

**Comment by fkiraly** [comment]
> How do I access dev-chat? I'll join discord

It is a channel on the sktime discord. Link in the readme (at the top, one of the badges)

**Comment by Apolo151** [comment]
I have worked on ClaSPTransformer #9133

**Comment by abdelrahman-omran** [comment]
I have worked on HIVECOTEV2 https://github.com/sktime/sktime/pull/9136

**Comment by Autumsjoy** [comment]
Hi @fkiraly, I'd like to work on adding a second test parameter set for OnlineEnsembleForecaster. I plan to use HedgeExpert for the second set and then remove the estimator from the exclusion list in _config.py

**Comment by vijith-bhima** [comment]
Hi @fkiraly! I'd like to work on adding a second test parameter set for CircularBinarySegmentation. Is it still available?

**Comment by AbdullahShheik** [comment]
Hi @fkiraly! I would like to work on SlopeTransformer.

**Comment by shreeya1102** [comment]
Hi @fkiraly . I would like to work on adding a second test parameter set for LogTransformer. Is this still available?

**Comment by archittmittal** [comment]
Hi @fkiraly, I would like to work on adding a second test parameter set for `OptionalPassthrough` and `DontUpdate`. I'll also remove them from the exclusion list in `_config.py` once done.

**Comment by archittmittal** [comment]
I've submitted a PR for `OptionalPassthrough` and `DontUpdate`: https://github.com/sktime/sktime/pull/9816

**Comment by ArjavJain123** [comment]
Hi @fkiraly , I've added new test parameter sets for SupervisedTimeSeriesForest and removed it from the exclusion list. Here is the PR: #9931 

I initially tried to include a few more estimators, but I'm starting with this one to ensure my workflow is correct. I'll pick up more once this is reviewed!

**Comment by Muawiya-contact** [comment]
Hi @fkiraly! I’m Muawiya, a new contributor interested in the ESoC rolling applications. I'd love to help with the community goal of increasing test coverage. I’ll be working on adding a second test parameter set for PCATransformer. Will submit a PR shortly!

**Comment by snoopuppy582** [comment]
Hi @fkiraly, I would like to work on adding a second test parameter set for PAA.
I reproduced that PAA.get_test_params() currently returns only one parameter set, and I plan to keep the PR limited to adding a second lightweight parameter set and removing PAA from 	est_get_test_params_coverage exclusions.

**Comment by nyxZehra** [comment]
Hi! @fkiraly 
I'd like to work on adding a second test parameter set for DummyRegressor, if no one else is already working on it.

**Comment by OfficialAbhinavSingh** [comment]
Hi @fkiraly — I'd like to take `PaddingTransformer` for this.

It currently has no `get_test_params`, so it falls back to the base class
single empty set. I'll add two sets covering the inferred-max-length path
(defaults) and the user-specified fixed-length path with a non-zero
`fill_value`, and remove it from `EXCLUDED_TESTS_BY_TEST` in
`sktime/tests/_config.py`. It's not in `EXCLUDE_SOFT_DEPS`, so no change
needed there.

**Comment by OfficialAbhinavSingh** [comment]
Hi @fkiraly — following up on #10656, I'd like to take `DilationMappingTransformer` next.

Its `get_test_params` currently returns a single set, `[{"dilation": 2}]`. I'll add
`{"dilation": 1}` as a second set — the documented boundary ("a dilation of 1 means
no change") and the degenerate single-slice path in `_dilate_series` — and remove the
estimator from `EXCLUDED_TESTS_BY_TEST` in `sktime/tests/_config.py`. It's not in
`EXCLUDE_SOFT_DEPS`, so no change needed there.

**Comment by sharvil112003** [comment]
Hi @fkiraly ! I'd like to work on adding a second test parameter set for `SlidingWindowSegmenter`.

**Comment by NAME-ASHWANIYADAV** [comment]
Hi @fkiraly - I do like to take the reduction forecasters in forecasting/compose/_reduce.py for this: DirRecTabular, DirRecTimeSeries, DirectTimeSeries, MultioutputTabular, MultioutputTimeSeries, RecursiveTabular, and RecursiveTimeSeries RegressionForecaster.

They all inherit the same get_test_params from _Reducer, which currently returns a single set, so one change to the shared method covers all of them consistently. I'll add a second set using a different regressor and window length, and remove the entries from EXCLUDED_TESTS_BY_TEST.

One question while I'm in there: DirectTabularRegressionForecaster is currently in EXCLUDE_SOFT_DEPS because its extra two parameter sets are gated on skpro. Once the shared method returns two unconditional sets it no longer needs that carve-out, so I plan to remove it from both lists. Let me know if you'd rather I left it alone

**Comment by OfficialAbhinavSingh** [comment]
Hi @fkiraly, I'd like to take `RandomSamplesAugmenter` for this.

It has no `get_test_params` of its own, so it falls back to the base class and returns a single empty dict. I'd add `{"n": 1.0, "without_replacement": True}` (the current defaults, so `create_test_instance` is unchanged) and `{"n": 5, "without_replacement": False}`.

The second set covers two different branches in `_transform`: an int `n` skips the `ceil(n * len(X))` fraction path, and `without_replacement=False` flips `replace` in `rng.choice`. With both sets I get 80 checks passing, against 47 today with `test_get_test_params_coverage` failing. I'd also remove it from `EXCLUDED_TESTS_BY_TEST`.

**Comment by NAME-ASHWANIYADAV** [comment]
Hi @fkiraly - I'll pick up PAAlegacy, SAXlegacy, and Hidalgo next.

I'll add a second parameter set for each (like num_intervals variants for PAA, K=2 for Hidalgo, and a new word length/alphabet size for SAX) and remove them from EXCLUDED_TESTS_BY_TEST.

I deliberately left the rest of the random-interval family alone since their tests are currently skipped (#3103), so adding parameters there would just be dead code for now.

**Comment by NiliRahmani** [comment]
Hi @fkiraly! I'd like to work on `SummaryClassifier` for this issue.

Its `get_test_params` currently returns a single dict for the unconditional
parameter set, so it's excluded from `test_get_test_params_coverage` via
`EXCLUDED_TESTS_BY_TEST` in `sktime/tests/_config.py`.

I'll add a second set using different summary functions with
`summary_quantiles=None`, so the quantile-free path is covered too, and remove
the estimator from the exclusion list. I'll leave the `results_comparison` set
unchanged, per the recipe above.

**Comment by coding-cosmos** [comment]
Hi @fkiraly, I'd like to work on the `ClearSky`

**Comment by NAME-ASHWANIYADAV** [comment]
Hi @fkiraly - next I'd like to take ElasticEnsemble and DistFromAligner.

ElasticEnsemble returns a single unconditional set. I'll add a second one with a different elastic distance (msm), probability-weighted voting instead of majority voting, and a subsampled train set in the parameter search, keeping the proportions small so both sets stay fast.

DistFromAligner currently returns a single set either way - {"aligner": AlignerDTW()} when dtw-python is installed, {} otherwise. I'll make it two unconditional sets using the dependency-free AlignerLuckyDtw (the same combination dists_kernels.lucky.DtwLucky already wraps internally) plus the default None aligner, and keep the AlignerDTW set as a conditional third when dtw-python is present. Both come out of EXCLUDED_TESTS_BY_TEST; neither is in EXCLUDE_SOFT_DEPS.

**Comment by AanchalGupta1162** [comment]
Hi @fkiraly, can I work on adding second test params for DOBIN?

**Comment by Harshavardhan-28** [comment]
Hi @fkiraly, can I work on adding test parameters for TimeSeriesKMedoids?

**Comment by maxelliotmingail-afk** [comment]
I would like to take FeatureSelection. Its current test params cover only method="all"; I will add a second fast, deterministic set using method="random", n_columns=1, and random_state=0, then remove FeatureSelection from the test_get_test_params_coverage exclusion list. This exercises the random column-selection branch without adding a soft dependency.

**Comment by NAME-ASHWANIYADAV** [comment]
Hi @fkiraly - next I'd like to take the dictionary-based family: SFAFast, IndividualBOSS, and IndividualTDE.

SFAFast returns a single dict; I'll add a second set varying the word/alphabet sizes, per-window normalisation, equi-width binning, and bigrams, with the same output-format parameters as the first set. IndividualBOSS and IndividualTDE have no get_test_params at all and fall back to the single empty base set; I'll add two sets each - the defaults as the first set (so create_test_instance is unchanged) and a second set covering the branches the defaults never reach (euclidean vs boss distance and chi2 feature selection for BOSS; spatial pyramid levels, IGB binning, and bigrams-off for TDE). All three come out of EXCLUDED_TESTS_BY_TEST; none is in EXCLUDE_SOFT_DEPS.

I checked SFA as well and left it out deliberately - it carries tests:skip_all=True, so parameter sets there would be dead code until that skip is lifted.

**Comment by NAME-ASHWANIYADAV** [comment]
Hi @fkiraly - next I'd like to take the dictionary-based ensembles: ContractableBOSS, MUSE, and WEASEL.

All three return a single unconditional dict. I'll add a second set to each - random feature selection and a widened window grid for ContractableBOSS, first-order differences plus bigrams for MUSE, and equi-depth binning plus bigrams for WEASEL - keeping the existing sets unchanged as the first entries and keeping everything small enough that the new sets fit in well under a second on the test scenarios. All three come out of EXCLUDED_TESTS_BY_TEST; none is in EXCLUDE_SOFT_DEPS. WEASEL's unrelated EXCLUDED_TESTS entry (test_multiprocessing_idempotent, [#5658](https://github.com/Nischal1425/sktime/issues/5658)) is left untouched.

I'm deliberately not touching TemporalDictionaryEnsemble yet - its get_test_params lives in the same file as IndividualTDE, which #10839 is currently modifying, and I'd rather not create a conflict between two of my own open PRs.

**Comment by NAME-ASHWANIYADAV** [comment]
Hi @fkiraly - next I'd like to take ShapeletTransform, ShapeletTransformClassifier, and TEASER.

The legacy ShapeletTransform has no get_test_params at all (the one in its file belongs to RandomShapeletTransform, which already has two sets) - I'll add two sets, the defaults plus a bounded-search configuration. ShapeletTransformClassifier and TEASER each return a single dict - I'll add a single-tree-estimator set for STC and a two-classification-points set for TEASER, exercising the multi-point early-classification path. All three come out of EXCLUDED_TESTS_BY_TEST; none is in EXCLUDE_SOFT_DEPS. STC's results_comparison set and TEASER's five known-issue EXCLUDED_TESTS entries are left untouched.

**Comment by oberoir080** [comment]
Hi @fkiraly, I'd like to take `TSFreshClassifier`. Planning to add a second parameter set with `relevant_feature_extractor=True`, since both current sets use `False` and `TSFreshRelevantFeatureExtractor` is never constructed in tests.

(Since, it's my first contribution. I wanted to keep it simple. Please let me know if it sounds good!)

**Comment by Sharon-study** [comment]
I’d like to work on HolidayFeatures.
It currently has only one test parameter set when the optional dependency is unavailable. I’ve reproduced the test_get_test_params_coverage failure locally and did not find an open PR addressing this.

**Comment by shubham5080** [comment]
Hi — I'll add second unconditional `get_test_params` sets for `TemporalDictionaryEnsemble` and `RandomIntervalSpectralEnsemble`, and remove their `test_get_test_params_coverage` skip tags. Both still return a single default dict on current `main`. PR shortly.
